### PR TITLE
Improve jvm-build-service integration

### PR DESCRIPTION
**Pull request by Michkov**
- add self-binding to workspace to have SystemConfig CRD available
- set SkipDryRunOnMissingResource=true to wait on SystemConfig CRD
- use one APIExport

### CODE REVIEW

File: components/hacbs/jvm-build-service/base/kustomization.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 resources:
 - https://github.com/redhat-appstudio/jvm-build-service/deploy/kcp?ref=954c86b39a12820c7910c93b1e20b9b572232816
+- self-binding.yaml
```
Comment (gabemontero) [inline components/hacbs/jvm-build-service/base/kustomization.yaml:3]: let's change the commit ^^ to `2bda08e7d8ff0b3d78c6564f54f183208b44af18` to pull in fixes from https://github.com/redhat-appstudio/jvm-build-service/pull/270

Comment (gabemontero) [reply components/hacbs/jvm-build-service/base/kustomization.yaml:3]: or should we replicate https://github.com/redhat-appstudio/build-service/pull/65 in jvm-build-service first and then move to that commit ?

File: components/hacbs/jvm-build-service/base/kustomization.yaml [hunk 2]
```diff
@@ -11,21 +12,33 @@ namespace: jvm-build-service
...
+      path: /metadata/name
+      value: jvm-build-service
+# Add APIExport flag to the operator arguments list
+- target:
+    kind: Deployment
+    name: hacbs-jvm-operator
+  patch: |-
+    - op: add
+      path: /spec/template/spec/containers/0/args/-
+      value: --api-export-name=jvm-build-service
```
Comment (gabemontero) [inline components/hacbs/jvm-build-service/base/kustomization.yaml:36]: https://github.com/redhat-appstudio/jvm-build-service/pull/282 should remove the need for this patch

**Review by gabemontero — COMMENTED** [review]
also, to reiterate what is in our slack thread today, I was unable to `oc apply -k ./deploy/kcp` and create the systemconfig with just the CRDs, and no apiexport, using the kcp admin config

File: components/hacbs/jvm-build-service/base/kustomization.yaml (hunk 1 above)
Comment (Michkov) [reply components/hacbs/jvm-build-service/base/kustomization.yaml:5]: going to update jvm-build-service to update automatically.

File: components/hacbs/jvm-build-service/base/self-binding.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: apis.kcp.dev/v1alpha1
+kind: APIBinding
+metadata:
+  name: jvm-build-service
+spec:
+  reference:
+    workspace:
```
[the remark this answers is not in the document]
Comment (Michkov) [reply components/hacbs/jvm-build-service/base/self-binding.yaml:8]: hash replacement will be part of next PR

File: components/hacbs/jvm-build-service/base/self-binding.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+      exportName: jvm-build-service
```
[the remark this answers is not in the document]
Comment (Michkov) [reply components/hacbs/jvm-build-service/base/self-binding.yaml:9]: Updated to accept standard types in this PR

**Pull request merged**